src/bootstrap/doc.rs

```diff
@@ -197,11 +197,21 @@ impl Step for TheBook {
         let compiler = self.compiler;
         let target = self.target;
 
+        let absolute_path = builder.src.join(&relative_path);
+        let redirect_path = absolute_path.join("redirects");
+        if !absolute_path.exists()
+            || !redirect_path.exists()
+            || dir_is_empty(&absolute_path)
+            || dir_is_empty(&redirect_path)
+        {
```

Comment:
We can create function something like `fn is_submodule_checked_out(module_path: &str) -> bool` that uses `git submodule foreach` command inside.